pwiz_tools/Skyline/Util/Adduct.cs

```diff
@@ -213,7 +213,7 @@ private Adduct(int charge, bool protonated)
                 {
                     // No leading + or - : is it because description starts with a label, or because + mode is implied?
                     var limit = input.IndexOfAny(new[] { '+', '-', ']' });

```

Comment:
I think this should maybe be:
var limit = input.IndexOfAny(new[] { '+', '-', ']' }, **posNext**);

You pretty much want to ignore any sign that is before the "M". But, if there is a sign somewhere before the M, and also after the M, you'd want to be able to find the sign after the M, right?

Reply:
Thanks, as usual, for a thoughtful review.
If there's a sign before the M it's not a well formed adduct description, so we'd rather know about that.